Source/WebCore/bindings/js/JSElementCustom.cpp

```diff
@@ -81,4 +85,62 @@ JSValue toJSNewlyCreated(JSGlobalObject*, JSDOMGlobalObject* globalObject, Ref<E
     return createNewElementWrapper(globalObject, WTFMove(element));
 }
 
+static JSValue getElementsArrayAttribute(JSGlobalObject& lexicalGlobalObject, const JSElement& thisObject, const QualifiedName& attributeName)
+{
+    auto& vm = JSC::getVM(&lexicalGlobalObject);
+    auto throwScope = DECLARE_THROW_SCOPE(vm);
+
+    JSObject* cachedObject = nullptr;
+    JSValue cachedObjectValue = thisObject.globalObject()->getDirect(vm, builtinNames(vm).cachedAttrAssociatedElementsPrivateName());
```

Comment:
Someone who knows better about JSC should comment whether this get & later put has any side effect or not.

@Constellation ?

Reply:
@mrego I think this will result in a situation where the cache is overwritten if accessed for different elements, because the private name is being stored on the global object rather than the element itself. I think `thisObject.getDirect()` and `thisObject.putDirect()` is what we want here

Reply:
@caitp thanks for the review, this was actually overwriting the cache if we were using different elements (I've added a new test case to verify that, that was failing with the original patch).

I've just uploaded a new version, storing it directly in the `JSElement` with `thisObject.get|putDirect()` as you suggests. And that fixes the new test case.

Let's see what other think about this.

Reply:
Internally, it can cause Structure transition, but I don't think it can be a user-observable side-effect so long as we are using private symbols.